corehq/ex-submodules/casexml/apps/phone/caselogic.py

```diff
@@ -51,13 +48,23 @@ def related(case_db, case):
           for case in initial_cases]
     )
     case_db.populate(directly_referenced_indices)
+
+    def process_queue():
+        new_relations = set()
+        while queue:
+            case = queue.pop()
+            if case and case['_id'] not in relevant_cases:
+                relevant_cases[case['_id']] = case
+                if case['doc_type'] == 'CommCareCase-Deleted':
+                    relevant_deleted_case_ids.append(case['_id'])
+                new_relations.update(index['referenced_id'] for index in indices(case))
+
+        if new_relations:
+            case_db.populate(new_relations)
```

Comment:
Do you know what this does if the case is already in case_db? Ah, checking the code it looks like it properly skips anything that's already in the case_db. (yes)